pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,227 @@
+
+Structure of the miniseries:
+* **Part 1: Designing the API's with AsyncAPI**
+* Part 2: Implementing the applications using code generation
+* Part 3: Black-box testing the applications using code generation
+* Part 4: Introducing new changes when using code generation
+* Part 5: The path to 1 billion players - Scaling the applications and finding bottlenecks with code generation
+
+**Don't see this blog post series as anything other then an example workflow, this is purely how I do it with my applications and how I use AsyncAPI and it's tooling to my advantage. Use this as an inspiration to finding an approach that works for you.**
+
+# Backstory
+
+Back in 2019 when I started contributing to the tooling of AsyncAPI, I was still in university studying for a masters in software engineering, and had at that point been a student developer at a company called [EURISCO](http://eurisco.dk/), for about 3 years. Beside that I have always had side projects that I worked on in my spare time, and it was one of these side projects that sparked my need for AsyncAPI. 
+
+My side project at that time was a [Rust](https://rust.facepunch.com/) game server plugin which collected in game events, such as when a player farms resources, kills another player, loots a container, etc, and send them to a backend. Later these could be extracted by an API to display the players progression and detailed account on what you did on the game server. 
+
+Initially I used OpenAPI to describe the API and the great thing was their tooling allowed me generate clients and servers in different languages which really speed up the implementation process.
+
+I soon encountered a use case that required me to push data to the game server, and solving this with REST was possible, but cumbersome. So I started exploring different alternatives, in terms of event driven architecture, however, none could be described using OpenAPI or offered tooling, so I had to find alternatives. 
+
+That was when I vaguely remembered a meeting where AsyncAPI was mentioned. Around that time, at work, we began to switch from a custom socket protocol to [NATS](https://nats.io/), and at the same time they spend some time figuring out how to mainstream the process and document the API's, and this is where they found and adopted AsyncAPI. So I started to use AsyncAPI in my project, and sparked my first ever encounter with open source, but that is a story for another time, maybe.
+
+So this blog post is a dedication to that experience, showcasing how I use AsyncAPI to document and generate code to speed up the development process, and maybe spark your interest in building the best tooling possible.
+
+# To that end
+
+Explaining something is always better with actual examples, therefore I will be creating a little system to show you how code generation can help the development process. 
+
+![General setup](/img/posts/jonaslagoni-miniseries-part1/blog-miniseries-general-setup.webp)
```

Comment:
you could use `<Figure>` component here and add nice caption to the image with content that you have below:

`The round dot between some broker and the game server represent how others may grab/interact with the application, ergo its API. The same thing is for the processor.`

this way reader can quickly understand those dots exactly when needed, without a need to read entire paragraph first

Reply:
Added 👍